### DISCUSSION TITLE
foundry-zksync compiler error

### DISCUSSION THREAD

**Discussion by Iammrgemini** [Cyfrin/foundry-full-course-cu#2812] [Q&A]
I used the instructions on this link https://docs.zksync.io/build/tooling/foundry/installation and https://foundry-book.zksync.io/ to get my foundryup and foundryup-zksync working just as it was on the course. but now when i try to compile my first zksync project,  i get the following error in my shell
```

[⠃] Using zksolc-1.5.4
[⠰] Compiling (zksync)
[⠔] Compiling 21 files with zksolc and ZKsync solc 0.8.28
Error: 
Broken pipe (os error 32)

````
i tried to see if there was a problem and there seems to some form of configurations in foundry.toml or perhaps somewhere i have no idea. i really need an explanation and how to get get pass this error. N.B: i am working with  a bash shell
<img width="960" alt="Code_KBj0EYjqq7" src="https://github.com/user-attachments/assets/f0fc8c0b-abb5-4253-a288-4acb68cf5468">

**Comment by EngrPips** [comment]
Hello @Iammrgemini, Please read this https://github.com/matter-labs/foundry-zksync/issues/453 discussion. It may help you.

**Reply by Iammrgemini** [reply]
Still does not solve my problem,  I don't think mine has a problem with the solc version as the error was only related to Broken pipe( os error 32)  it seems as the threat is currently closed and no new update has been made on it

**Reply by EngrPips** [reply]
I will advise you to report your issue on the official Foundry-zksync repo and await a response from the team.

**Reply by neerajnerlekar** [reply]
I had the same issue of `Broken pipe` and @EngrPips linked article helped resolve it.

I changed the `pragma solidity ^0.8.24` to `pragma solidity 0.8.24`, so the it compiles for 0.8.24 instead of 0.8.28.

```bash
lotus:~/foundry/merkle-airdrop$ forge build --zksync
[⠃] Using zksolc-1.5.4
[⠊] Compiling (zksync)
[⠒] Compiling 61 files with zksolc and ZKsync solc 0.8.28
Error: Broken pipe (os error 32)
lotus:~/foundry/merkle-airdrop$ forge clean
lotus:~/foundry/merkle-airdrop$ forge build --zksync
[⠃] Using zksolc-1.5.4
[⠒] Compiling (zksync)
[⠢] Compiling 61 files with zksolc and ZKsync solc 0.8.24
[⠆] zksolc and ZKsync solc 0.8.24 finished in 9.05s
Compiler run successful!
lotus:~/foundry/merkle-airdrop$
```

**Reply by EngrPips** [reply]
Awesome, I am glad that helped you.

**Reply by Iammrgemini** [reply]
Yeah I changed mine to 0.8.20 and it worked. Forgot to update that i passed that phase on here. Thanks @EngrPips

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.
